forge-ai/src/main/java/forge/ai/AiAttackController.java

```diff
@@ -798,11 +800,37 @@ && isEffectiveAttacker(ai, attacker, combat, defender)) {
                     // TODO: if there are other ways to tap this creature (like mana creature), then don't need to attack
                     mustAttack = true;
                 } else {
-                    // TODO move to static Ability
-                    if (attacker.hasKeyword("CARDNAME attacks each combat if able.") || attacker.hasStartOfKeyword("CARDNAME attacks specific player each combat if able")) {
-                        // TODO switch defender if there's one without a cost or it's not the specific player
-                        mustAttack = true;
-                    } else if (attacker.getController().getMustAttackEntityThisTurn() != null && CombatUtil.getAttackCost(ai.getGame(), attacker, defender) == null) {
+                    final Game game = attacker.getGame();
+                    for (final Card ca : game.getCardsIn(ZoneType.STATIC_ABILITIES_SOURCE_ZONES)) {
+                        for (final StaticAbility stAb : ca.getStaticAbilities()) {
```

Comment:
should probably extract the common parts here and from AttackRequirement into own StaticAbility class

It will probably grow later when stuff like MustAttackEffect gets reworked in the same manner